docs/user/developing_jobs.md

```diff
@@ -4,224 +4,155 @@ A goal of this plugin is to make it relatively quick and straightforward to deve
+Regardless of which direction you are synchronizing data in, you need to write the `load` method for the remote adapter yourself. You can find many examples of how this can be done in the `nautobot_ssot.integrations` module, which contains pre-existing integrations with remote systems.
 
 !!! note
-    In Nautobot 2.0, jobs will no longer be atomic by default so this section will not apply anymore.
-
-#### Using Bulk ORM Operations
-
-Bulk ORM operations are available within Django to perform multiple inserts/updates in a single DB query. Take a look at the documentation for [bulk_create](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#bulk-create) for example. These bulk operations should only be used very carefully in Nautobot. The following caveats apply when using them:
+As the features in `nautobot_ssot.contrib` are still very new, most existing integrations do not use them.
 
-- No change log is generated
-- No model validation is performed (you could for example connect a cable to the same interface on both ends, causing UI views to break)
-- No custom `save` methods are called (for the `Device` model for example this causes all device components from the device type not to be created)
+### Step 3 - The Job
 
-Bulk ORM operations can nevertheless be a performance improvement, you just have to be very careful when employing them.
+Develop a Job class, derived from either the `nautobot_ssot.jobs.base.DataSource` or `DataTarget` classes provided by this plugin, and implement the adapters to populate the self.source_adapter and self.target_adapter that are used by the built-in implementation of sync_data. This sync_data method is an opinionated way of running the process including some performance data, more in next section, but you could overwrite it completely or any of the key hooks that it calls.
 
-## Analyzing Job Performance
+The methods [`calculate_diff`][nautobot_ssot.jobs.base.DataSyncBaseJob.calculate_diff] and [`execute_sync`][nautobot_ssot.jobs.base.DataSyncBaseJob.calculate_diff] are both implemented by default, using the data that is loaded into the adapters through the respective methods. Note that `execute_sync` will _only_ execute when dry-run is set to false.
 
-In general there are two different metrics to optimize for when developing SSoT jobs:
+Optionally, on your Job class, also implement the [`lookup_object`][nautobot_ssot.jobs.base.DataSyncBaseJob.calculate_diff], [`data_mapping`][nautobot_ssot.jobs.base.DataSyncBaseJob.calculate_diff], and/or [`config_information`][nautobot_ssot.jobs.base.DataSyncBaseJob.config_information] APIs (to provide more information to the end user about the details of this Job), as well as the various metadata properties on your Job's Meta inner class. Refer to the example Jobs provided in this plugin for examples and further details.
```

Comment:
Are the `lookup_object`, `data_mapping`, and `config_information` intended to be links? If so, then they should be wrapped in paranthesis. Also, it appears you use calculate_diff twice. Regarding Glenn's point, it might be best to instead use a permalink to the part of the code being referenced?